tests/driver/index.test.ts

```diff
@@ -355,6 +355,23 @@ describe(`Driver based tests`, async () => {
       );
     });
 
+    it('create collection with options', async () => {
+      const vectorOptions = {
+        "vector": {
+          "size": 1000,
+          "function": "cosine"
+        }
+      }
+
+      // @ts-ignore
+      const personSchema = new mongooseInstance.Schema({
+        name: String
+      }, vectorOptions);
```

Comment:
This is an interesting alternative approach, but I think the approach from #120 is more robust. 

```javascript
const vectorOptions = {
  "collectionOptions": { 
    "vector": {
      "size": 1000,
      "function": "cosine"
    }
  }
}
```

The approach of pulling `schemaUserProvidedOptions` works, but unfortunately `schemaUserProvidedOptions` isn't part of Mongoose's public API. That property is just an internal workaround for passing `bufferCommands` and `bufferTimeoutMS` to collections, and is subject to change.